src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -333,6 +333,7 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
         variant_index: VariantIdx
     ) {
         if self.layout.for_variant(bx.cx(), variant_index).abi.is_uninhabited() {
+            bx.unreachable();
```

Comment:
You probably need to start a new block, or stop the remaining MIR statements/terminator from being codegen'd.

cc @rkruppe @nagisa

Reply:
For now I made this an `abort`, also matching what we do for uninhabited returns. But it seems odd to me that we'd give those well-defined "trap" semantics as opposed to exploiting their UB.